onnx2kerastl/reshape_layers.py

```diff
@@ -174,6 +174,23 @@ def convert_reshape(node, params, layers, lambda_func, node_name, keras_name):
     input_1 = layers[node.input[1]]
 
     if is_numpy(input_1):
+
+        # check reshaping type and validity
+        contains_zero_dim = np.isin(input_1, 0).any()
+        contains_infer_dim = np.isin(input_1, -1).any()
+        dims_to_ignore = None
+        dims_to_keep_unchanged = None
+        if params['allowzero']:
+            if contains_infer_dim and contains_zero_dim:
+                raise ValueError(
+                    "Reshape parameter 'allowzero' is set and reshaping argument contains both '0' dim and '-1'"
+                    "which is not allowed"
+                    f"node name: {node_name}")
+            elif contains_zero_dim:
+                dims_to_ignore = np.argwhere(input_1 == 0)
+        elif not params['allowzero'] and contains_zero_dim:
+            dims_to_keep_unchanged = np.squeeze(np.argwhere(input_1 == 0))
```

Comment:
I'm not sure if that code fits the documentation of `allowzero` in onnx, maybe i miss understood  something, lets talk about

> allowzero : int (default is 0)
(Optional) By default, when any value in the 'shape' input is equal to zero the corresponding dimension value is copied from the input tensor dynamically. allowzero=1 indicates that if any value in the 'shape' input is set to zero, the zero value is honored, similar to NumPy.